packages/ui/src/components/pages/Accounts/AccountControls/ShowAddressQrCodeDialog.tsx

```diff
@@ -0,0 +1,58 @@
+import { QRCodeCanvas } from 'qrcode.react';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useEffectOnce } from 'react-use';
+import { encodeAddress } from '@polkadot/util-crypto';
+import { Dialog, DialogContent } from '@mui/material';
+import AccountAddress from 'components/pages/Accounts/AccountAddress';
+import DialogTitle from 'components/shared/DialogTitle';
+import NetworksSelection from 'components/shared/NetworksSelection';
+import useDialog from 'hooks/useDialog';
+import useQrCode from 'hooks/useQrCode';
+import { AccountInfoExt } from 'types';
+import { EventName, EventRegistry } from 'utils/eventemitter';
+
+export default function ShowAddressQrCodeDialog(): JSX.Element {
+  const { t } = useTranslation();
+  const { open, doOpen, doClose } = useDialog();
+  const { ref, size } = useQrCode();
+  const [address, setAddress] = useState<string>('');
+  const [name, setName] = useState<string>('');
+
+  const handleNetworkChange = (addressPrefix: number) => {
+    setAddress(encodeAddress(address, addressPrefix));
+  };
+
+  const onOpen = (account: AccountInfoExt) => {
+    setAddress(account.networkAddress);
+    setName(account.name!);
+    doOpen();
+  };
+
+  useEffectOnce(() => {
+    EventRegistry.on(EventName.OpenShowAddressQrCodeDialog, onOpen);
+    return () => {
+      EventRegistry.off(EventName.OpenShowAddressQrCodeDialog, onOpen);
+    };
+  });
+
+  const onClose = () => {
+    doClose(() => {
+      setAddress('');
+      setName('');
+    });
+  };
+
+  if (!address || !name) return <></>;
```

Comment:
We would want to show the dialog if `name` is empty. The purpose here is to show `address`, so it's completely fine to not show the name.